### ISSUE TITLE
Exercise 2 Streams (Iterate not working)

### ISSUE THREAD

**Issue by yewey2** [nus-cs2030/2223-s1#221]
I'm redoing exercise 2 with streams, but iterate doesn't seem to work...

![image](https://user-images.githubusercontent.com/83965035/197524002-140f2d9a-85c0-4feb-9b12-397893556bd4.png)

Does anyone have any hints on doing Exercise 2 without using streams?

**Comment by Ryan-loves-movies** [comment]
I imagine you're trying to convert the cruises list to a stream here -- "Stream.<Cruise>iterate(0, i -> i + 1).map(i -> cruises.get(i)).toList()" cruises.stream() should work

**Comment by yewey2** [comment]
@Ryan-loves-movies
No, I'm iterating through pairs and storing a state within the pairs. Any way around it?

**Comment by Ryan-loves-movies** [comment]
Would a Stream.<Pair>iterate(x->x).reduce(initpair, (firstPair, secPair) -> ...) work for your purposes?
